ext/Errno/Errno_pm.PL

```diff
@@ -271,9 +271,9 @@ sub write_errno_pm {
 	close(CPPO);
     }
 
-    # escape $Config{'archname'}
-    my $archname = $Config{'archname'};
-    $archname =~ s/([@%\$])/\\$1/g;
+    # escape $Config{'archname'}, $Config{'osvers'}
+    my ($archname, $osvers) = @Config{'archname', 'osvers'};
+    s/([@%\$])/\\$1/g for $archname, $osvers;
```

Comment:
`%` isn't special in double-quoted strings and doesn't need to be escaped. On the other hand, `"` and `\` probably should be escaped. (Or just `quotemeta` the whole thing?)

Reply:
Unfortunately adding " to archname or osvers breaks the build when compiling perl, since neither ARCHNAME nor OSVER is escaped in config_h.PL.  I expect `\` would have similar problems.

I've changed this use quotemeta, so if we fix `config_h.SH` these will be handled correctly.